aten/src/ATen/native/TensorShape.cpp

```diff
@@ -3757,22 +3757,59 @@ std::vector<Tensor> unflatten_dense_tensors(const Tensor& flat, TensorList tenso
   return outputs;
 }
 
+
+// Clones a tensor by cloning the underlying storage that it came from,
+// which allows us to replicate the exact strides/storage_offset in the cloned tensor.
+// Note [*_scatter ops preserve strides]
+// In order for functionalization to preserve stride correctness, the *_scatter
+// operators that it calls must preserve the striding behavior of their inputs.
+// Specifically, the output of *_scatter(base, mutated_view, ...)
+// should have identical size/stride/storage_offset to "base".
+at::Tensor clone_preserve_strides(const at::Tensor& self) {
```

Comment:
nit: maybe you want to use the _new_zeros_with_same_feature_meta function that we created for forward AD (it supports vmap).
We could add a flag to choose between filled with zeros vs clone.